pkg/web/apps/apps_repository.go

```diff
@@ -14,5 +14,12 @@ type Repository interface {
 	DeleteAppByAppID(appId string) error
 	CreateApp(id, appId, name string) error
 	GetAppByAppID(appID string) (*models.App, error)
+	GetActiveAppByAppID(appID string) (*models.App, error)
 	UnDeleteAppByAppID(appID string) error
+	GetVersionByAppIDAndVersion(appID string, versionNumber string) (*models.Version, error)
+	GetDeviceByDeviceIDAndAppID(deviceID string, appID string) (*models.Device, error)
+	GetDeviceByVersionAndAppID(versionID string, appID string) (*models.Device, error)
+	CreateNewVersion(version *models.Version) error
+	IncrementVersionTotals(versionID string, isNewInstall bool) error
+	InsertDeviceOrUpdateVersionID(device *models.Device) error
```

Comment:
See that all methods still indeed the `InsertDeviceOrUpdateVersionID `. Just one was changed for CreateNewVersion and was added the IncrementVersionTotals in order to delegate the count of total = total +1 to the DB and try to void concurrence issues.